google/cloud/internal/curl_impl.cc

```diff
@@ -114,15 +113,74 @@ class WriteVector {
       }
       std::copy(src.begin(), src.end(), dst.begin());
       dst.remove_prefix(src.size());
-      writev_.pop_back();
+      writev_.pop_front();
     }
     return avail - dst.size();
   }
 
+  /// Implements a CURLOPT_SEEKFUNCTION callback.
+  ///
+  /// @see https://curl.se/libcurl/c/CURLOPT_SEEKFUNCTION.html
+  /// @returns true if the seek operation was successful.
+  bool Seek(std::size_t offset, int origin) {
+    // libcurl claims to only req
+    if (origin != SEEK_SET) return false;
+    writev_.assign(original_.begin(), original_.end());
+    // Reverse the vector so the first chunk is at the end.
+    std::reverse(writev_.begin(), writev_.end());
```

Comment:
The reverse was originally so that `vector::pop_back()` was used to consume data, but with the move to `deque` that is less of a consideration.  Still, it seems (without looking deeply) that it should be all or none.

Reply:
Ooops I will send a fix in a few minutes.